### PR TITLE
BC-3010 - cleanup and update docs

### PR DESCRIPTION
**Pull request by hoeppner-dataport**
# Short Description
The developer-documentation in the Schulcloud was updated in many aspects:
- reflect Nuxt.JS removal and changes applied with the removal
- add BestPractices and decisions (e.g. from FrontendArcGroup)
- remove outdated information and deprecated instructions

## Links to Ticket and related Pull-Requests
https://ticketsystem.dbildungscloud.de/browse/BC-3010

## Checklist before merging
- [x] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] PO: Any deviation from requirements was agreed with Product-Owner / ticket author / support-team

> Notice: Please keep this Pull-Request as a Draft (or add WIP label), until it is ready to be reviewed

### CODE REVIEW

File: src/pages/administration/LDAPActivate.page.vue [hunk 1]
```diff
@@ -212,9 +212,6 @@ const redirectToConfigPage = (page) => {
 
 export default {
 	components: { ModalBodyInfo, ModalFooterConfirm, InfoMessage },
-	meta: {
-		requiredPermissions: ["ADMIN_VIEW", "SCHOOL_EDIT"],
```
Comment (virgilchiriac) [inline src/pages/administration/LDAPActivate.page.vue:216]: how is this now handled? Should it be part of this PR?

Comment (muratmerdoglu-dp) [reply src/pages/administration/LDAPActivate.page.vue:216]: No, It is not part of this ticket. It's some remaining nuxt-removal things that we weren't aware of. The permissions now are controlled with `permission.guard.ts` in router folder.

File: src/pages/ProxyError.page.vue [hunk 2]
```diff
@@ -15,9 +15,6 @@ export default {
 		ErrorPage,
 	},
 	layout: "loggedOut",
-	meta: {
-		isPublic: true,
```
Comment (arnegns) [inline src/pages/ProxyError.page.vue:19]: So there is no default anymore for this page. You only set this in router. Correct?

**Review by Loki-Afro — DISMISSED** [review]

File: src/pages/administration/LDAPActivate.page.vue (hunk 1 above)
Comment (hoeppner-dataport) [reply src/pages/administration/LDAPActivate.page.vue:216]: @virgilchiriac The way page-permissions work now is described here:
**3_HowTo.md** > Section **User-Permissions on Pages**

**Review by OliverHappe — APPROVED** [review]

**Comment by hoeppner-dataport** [comment]
@janrenz We could by postponing the update of the project-description in the main README.md. The approval of those and the rewriting by others and reapproval by others... was blocking the merge. Now this will be updated at some later time.

**Pull request merged**